### ISSUE TITLE
Gesture detectors are unable to capture small, detailed gestures.

### ISSUE THREAD

**Issue by MichaelFerrier** [flutter/flutter#33234]
My app uses a GestureDetector's onPanStart, onPanUpdate and onPanEnd handlers to allow the user to draw on the screen. This worked great when I first developed the app in late 2018, but after recently updating Flutter and recompiling the app, I found that GestureDetector is much less sensitive to small movements. 

By adding an onPanDown handler, I found that onPanDown is being called as soon as the pointer touches the screen, but then a fairly large drag movement is required before onPanStart will be called, and before onPanUpdate will then start being called for every ensuing movement of the pointer. 

Because of how large of a movement is required before the gesture is considered a "pan", my app is no longer sensitive to small movements, and so users would not be able to draw small details as they previously could.

After digging into the Flutter code, the only solution I could find was to edit this file:

flutter\packages\flutter\lib\src\gestures\constants.dart

...to change the constant kTouchSlop from 18.0 to a lower number such as 4.0. 

This is a messy solution and would need to be redone each time I update the Flutter code. So, I'm wondering if there's a better alternative method to allow GestureDetector (or ImmediateMultiDragGestureRecognizer, for multitouch gestures) to require less movement before it begins registering pans, and so to allow small, detailed gestures to be captured?   

If not, I recommend adding the ability to change kTouchSlop on a case by case basis.

-------------------------------------
Flutter doctor -v

[√] Flutter (Channel beta, v1.5.4-hotfix.2, on Microsoft Windows [Version 10.0.17134.765], locale en-US)
    • Flutter version 1.5.4-hotfix.2 at D:\src\flutter
    • Framework revision 7a4c33425d (3 weeks ago), 2019-04-29 11:05:24 -0700
    • Engine revision 52c7a1e849
    • Dart version 2.3.0 (build 2.3.0-dev.0.5 a1668566e5)

[√] Android toolchain - develop for Android devices (Android SDK version 28.0.3)
    • Android SDK at C:\Users\frogr\AppData\Local\Android\sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-28, build-tools 28.0.3
    • Java binary at: D:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1136-b06)
    • All Android licenses accepted.

[√] Android Studio (version 3.2)
    • Android Studio at D:\Program Files\Android\Android Studio
    • Flutter plugin version 30.0.1
    • Dart plugin version 181.5656
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1136-b06)

[√] Connected device (1 available)
    • SAMSUNG SGH I337 • c65ac68a • android-arm • Android 5.0.1 (API 21)

• No issues found!

**Comment by TahaTesser** [comment]
Hi @MichaelFerrier 
Are you still experiencing this issue?
Thank yo

**Comment by MichaelFerrier** [comment]
Yes, I'm still experiencing this. I recently upgraded to Flutter 1.14.6, and my app was once again unable to capture fine details that the user draws due to the problem described above. 

Again I was able to fix this by changing kTouchSlop from 18.0 to 4.0. This is not an ideal solution; it requires modifying the flutter code after each upgrade, and applies globally so it makes other operations too sensitive (such as making it easy to accidentally swipe horizontally from one tab to another when trying to swipe vertically through a list). Thank you.

**Comment by iapicca** [comment]
Hi @MichaelFerrier 
like in this similar (not identical) [issue](https://github.com/flutter/flutter/issues/18554)
would making this issue a feature request for customisable touch slop
solve the problem?
thank you

**Comment by MichaelFerrier** [comment]
Hi @iapicca 
Yes, a good solution would be to allow the default touch slop to be over ridden on a per-widget basis. That way widgets could be set to capture small detailed gestures, as in my case of using GestureDetector to allow the user to draw on the screen.
Thank you,
Michael

**Comment by ABleas** [comment]
I'd like to add that I'm also experiencing this issue. I am adding a pinch-zoom mechanic to change the zoom of a camera. But it feels unresponsive and not as accurate as the camera app usually provided on many devices.

A customisable touch slop would also be a good solution for me.

**Comment by petermakeswebsites** [comment]
I also have a good use case for per-widget kTouchSlop managing. It would be really nice to have this feature.

**Comment by lukepighetti** [comment]
I have experience this in the past, but noticed it very clearly in a Flame side project where it's almost impossible to have fine initial control of an on-screen joystick because of this issue.

https://user-images.githubusercontent.com/666539/204091320-6361b997-a4b7-4f25-a25c-55284720bb00.mp4

**Comment by iapicca** [comment]
@TahaTesser 
I think the labels as `proposal` `new feature` I added 1 million years ago are wrong (sorry guys)
what @lukepighetti is describing looks like bug to me

**Comment by ritaxbiz** [comment]
Hey guys, any update on this? Currently it's not able to draw a small line, and the drawing feature is not smooth. Or do you guys have any alternative on this when implementing a drawing feature?

**Comment by sreeraj-gs** [comment]
same issue here can't even draw a small circle due to this, is there any recognize without this delay?

DragStartBehavior.down
```
flutter:  onTapDown:
flutter: Pencil Handler onTapDOWN: at Offset(873.0, 2210.1)
flutter:  onPanStart:
flutter: WBCanvas: onPanStart called
flutter: Pencil Handler onPanStart: at Offset(873.0, 2210.1)
flutter:  onPanUpdate:
flutter: WBCanvas: onPanUpdate called
flutter: Pencil Handler onPanUpdate: at Offset(907.5, 2210.6)
flutter:  onPanUpdate:
flutter: WBCanvas: onPanUpdate called
flutter: Pencil Handler onPanUpdate: at Offset(911.0, 2210.6)
```

**Comment by Sreeraj9764** [comment]
same issue here can't even draw a small circle due to this, is there any recognizer without this delay?

DragStartBehavior.down

```
flutter:  onTapDown:
flutter: Pencil Handler onTapDOWN: at Offset(873.0, 2210.1)
flutter:  onPanStart:
flutter: WBCanvas: onPanStart called
flutter: Pencil Handler onPanStart: at Offset(873.0, 2210.1)
flutter:  onPanUpdate:
flutter: WBCanvas: onPanUpdate called
flutter: Pencil Handler onPanUpdate: at Offset(907.5, 2210.6)
flutter:  onPanUpdate:
flutter: WBCanvas: onPanUpdate called
flutter: Pencil Handler onPanUpdate: at Offset(911.0, 2210.6)
```
